### PR TITLE
Bugfixes

### LINKED ISSUE #25: Checkbox is on incorrect side of AWS icon
Looks like the checkbox for AWS (in hosting infrastructure) should be on the left, but is actually on the right side.

screenshot 2018-01-26 14 13 55

@rdebeasi can you please have a look?

### PR DESCRIPTION
**Pull request by rdebeasi**
- On the reference diagram, OpenShift dedicated is now greyed out if it's not selected.
- Validate project name in the UI - only lowercase letters, numbers, and dashes are allowed.
- Fix Ansible logo on reference diagram. (In Chrome, this would sporadically fail to appear.)
- Make styled checkboxes work on browsers other than Chrome and Safari.
- When a validation error occurs, scroll the error message into view so that the user knows what happened.
- Improve performance of the "internal" version of the page.
- Consolidate confusing "locked" and "grayed out" states into one "locked" state. (This is an internal code tweak; the checkboxes themselves look the same.)

The Chrome + Ansible logo bug was a race condition. `internal.html` loads content from `index.html`, and if the JavaScript layout code runs before the content has finished loading, layout problems appear. To fix that problem, I switched to using index.html for both versions of the page and accessing the "internal" version of the page via a query string (e.g. `localhost:8080/?internal=true`). There's still an `internal.html` file, but it's there solely to redirect old URLs to their new equivalents.

resolves #25 

@oybed, do these changes look okay to you?
CC @mwalker5000

### CODE REVIEW

File: website/js/app.js [hunk 1]
```diff
@@ -396,6 +393,12 @@ function callStack(projName, username, gitRepo) {
                 }
             }
         });
+        $('#projectName').on('input', function() {
+          this.setCustomValidity('');
+          if (!this.validity.valid && this.value !== '') {
+            this.setCustomValidity('Please use only lowercase letters, numbers, and dashes.');
```
Comment (oybed) [inline website/js/app.js]: The message looks good, but any chance we can also turn the input box "red" when the focus is shifted away from the box?

File: website/js/app.js [hunk 2]
```diff
@@ -3,6 +3,13 @@ $(document).ready(function() {
     // init foundation
     $(document).foundation()
 
+    // If this flag is set, allow the user to kick off an Ansible job
+    window.rhtLabsInternal = (getQueryVariable('internal') === 'true');
+    if (window.rhtLabsInternal) {
+      // Persist internal flag when form is submitted.
+      $('#form').append('<input type="hidden" name="internal" value="true" />');
+    }
+
```
Comment (oybed) [inline website/js/app.js:12]: There's also a check at the bottom of this file for `rhtLabsInternal` and sets the form action to `internal.html` - shouldn't that now be updated? I know `internal.html` redirects, but probably best to eliminate if possible?

**Review by oybed — CHANGES_REQUESTED** [review]
Overall looks good. A couple of comments/requests inline. We'll have to find a time that works to test in the live system

File: website/js/app.js (hunk 2 above)
Comment (rdebeasi) [reply website/js/app.js:12]: Thanks for catching that! Fixed.

File: website/js/app.js (hunk 1 above)
Comment (rdebeasi) [reply website/js/app.js]: Sure! I've made that change.

**Review by oybed — APPROVED** [review]
Looks good. Merging!

**Pull request merged**